src/editor/nodes/TroikaTextNode.js

```diff
@@ -0,0 +1,219 @@
+// This is a port of https://github.com/protectwise/troika/tree/master/packages/troika-three-text
+// that has been made functional and restyled for Mozilla Hubs and Spoke (hubs.mozilla.com)
+// by @jamesckane at Paradowski Creative (paradowski.com)
+
+import { Color, Object3D } from "three";
+import EditorNodeMixin from "../../../../src/editor/nodes/EditorNodeMixin";
+import { Text } from "troika-three-text";
+
+export default class TroikaTextNode extends EditorNodeMixin(Object3D) {
+  static legacyComponentName = "troika-text";
+
+  static nodeName = "Troika Text";
+
+  constructor(editor) {
+    super(editor);
+    this.editor = editor;
+    this.scene = editor.scene;
+    this.text = "Text";
+    this.anchorX = "left";
+    this.anchorY = "top";
+    this.color = new Color();
+    this.curveRadius = 0;
+    this.depthOffset = 0;
+    this.direction = "auto";
+    this.fillOpacity = 1;
+    this.font;
+    this.fontSize = 10;
+    this.textAlign = "left";
+    this.letterSpacing = 0;
+    // this.clipRect = "";  // Todo, implement later
+    this.lineHeight = 1;
+    this.outlineBlur = 0;
+    this.outlineColor = new Color();
+    this.outlineOffsetX = 0;
+    this.outlineOffsetY = 0;
+    this.outlineOpacity = 1;
+    this.outlineWidth = 0;
+    this.overflowWrap = "normal";
+    this.strokeColor = new Color();
+    this.strokeOpacity = 1;
+    this.strokeWidth = 0;
+    this.textAlign = "left";
+    this.textIndent = 0;
+    this.whiteSpace = "normal";
+    this.maxWidth = 9999; // The serialize method can't seem to handle a value of Infinity, just feed it ~10k meters
+    this.troikaText = new Text();
+  }
+
+  onChange(change) {
+    // change is undefined when serialize returns - but whole node has been updated
+    // change will be named if coming from a UI input
+    if (change === "position" || change === "scale" || change === "rotation") return;
+    else if (change === "color" || change === "strokeColor" || change === "outlineColor") {
+      this.troikaText[change] = new Color(this[change]);
+    } else this.troikaText[change] = this[change];
+    this.troikaText.sync();
+  }
+
+  onRemove() {
```

Comment:
We can probably avoid `onAdd` and `onRemove` if we inherit from `Text`.

Reply:
These are now handled by the `this.helper` convention used throughout the Spoke project.